nilearn/decoding/searchlight.py

```diff
@@ -408,8 +409,9 @@ def fit(self, imgs, y, groups=None):
 
         # Get the seeds
         if self.mask_img is not None:
-            self.mask_img = check_niimg_3d(self.mask_img)
-        process_mask_img = self.process_mask_img or self.mask_img
+            self.mask_img_ = deepcopy(self.mask_img)
+            self.mask_img_ = check_niimg_3d(self.mask_img_)
```

Comment:
Should `self.mask_img_` be set to `None` if `self.mask_img` is `None`?

Reply:
yeah that's probably better: will fix

Reply:
done